tests/agent/test_agent.py

```diff
@@ -98,3 +104,164 @@ async def test_read_agent_config_container_2valid(arpcs_no_ainit, mocker):
 
     assert arpcs_no_ainit.local_config.container.kernel_gid.real == 10
     assert arpcs_no_ainit.local_config.container.kernel_uid.real == 20
+
+
+@pytest.mark.asyncio
+async def test_update_scaling_group_changes_config() -> None:
+    mock_agent = Mock(spec=DummyAgent)
+    mock_agent.local_config = AgentUnifiedConfig(
+        agent=AgentConfig(backend=AgentBackend.DUMMY, scaling_group="default"),
+        container=ContainerConfig(scratch_type=ScratchType.HOSTDIR),
+        resource=ResourceConfig(),
+        etcd=EtcdConfig(namespace="test", addr=HostPortPair(host="127.0.0.1", port=2379)),
+    )
+
+    AbstractAgent.update_scaling_group(mock_agent, "gpu")
+
+    assert mock_agent.local_config.agent.scaling_group == "gpu"
+
+
+@pytest.mark.asyncio
+async def test_update_scaling_group_persists_single_agent(tmp_path) -> None:
+    config_file = tmp_path / "agent.toml"
+    config_file.write_text(
+        """[agent]
+backend = "dummy"
+scaling-group = "default"
+id = "test-agent"
+
+[container]
+scratch-type = "hostdir"
+
+[resource]
+
+[etcd]
+namespace = "test"
+addr = { host = "127.0.0.1", port = 2379 }
+"""
+    )
```

Comment:
Please separate the necessary parts for setup into fixtures, and group the test codes based on TestClass.